In [5]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt

In [6]:
education_districtwise = pd.read_csv('C:/Users/pmari/Downloads/Python Goggle Advanced/education_districtwise.csv')


Explore the data
Let's start with the head()function to get a quick overview of the dataset. Recall that head() will return as many rows of data as you input into the variable field.


In [7]:
education_districtwise.head(10)

DISTNAME STATNAME  BLOCKS  VILLAGES  CLUSTERS  TOTPOPULAT  OVERALL_LI
0   DISTRICT32   STATE1      13       391       104    875564.0       66.92
1  DISTRICT649   STATE1      18       678       144   1015503.0       66.93
2  DISTRICT229   STATE1       8        94        65   1269751.0       71.21
3  DISTRICT259   STATE1      13       523       104    735753.0       57.98
4  DISTRICT486   STATE1       8       359        64    570060.0       65.00
5  DISTRICT323   STATE1      12       523        96   1070144.0       64.32
6  DISTRICT114   STATE1       6       110        49    147104.0       80.48
7  DISTRICT438   STATE1       7       134        54    143388.0       74.49
8  DISTRICT610   STATE1      10       388        80    409576.0       65.97
9  DISTRICT476   STATE1      11       361        86    555357.0       69.90

📘 Nota sobre la interpretación de los datos
Para interpretar estos datos correctamente, es importante entender que cada fila, u observación, se refiere a un distrito diferente (y no, por ejemplo, a un estado o a una aldea).

La columna VILLAGES indica cuántas aldeas hay en cada distrito.

La columna TOTPOPULAT indica la población de cada distrito.

La columna OVERALL_LI indica la tasa de alfabetización de cada distrito.

🧮 Uso de describe() para calcular estadísticas descriptivas
Ahora que tenemos una mejor comprensión del conjunto de datos, vamos a usar Python para calcular estadísticas descriptivas.

Cuando se calculan estadísticas descriptivas en Python, la función más útil es describe().
Los profesionales de datos la utilizan como una forma práctica de calcular muchas estadísticas clave de una sola vez.

Para una columna numérica, describe() devuelve:

count: número de observaciones no nulas (no NA/null).

mean: promedio aritmético.

std: desviación estándar.

min: valor más pequeño (mínimo).

25%: primer cuartil (percentil 25).

50%: mediana (percentil 50).

75%: tercer cuartil (percentil 75).

max: valor más grande (máximo).

📖 Referencia: pandas.DataFrame.describe

🎯 Enfoque principal: tasa de alfabetización
Nuestro principal interés es la tasa de alfabetización, contenida en la columna OVERALL_LI, que muestra la tasa de alfabetización de cada distrito del país.
Usa la función describe() para revelar las estadísticas clave sobre la tasa de alfabetización.

In [8]:
education_districtwise['OVERALL_LI'].describe()

count    634.000000
mean      73.395189
std       10.098460
min       37.220000
25%       66.437500
50%       73.490000
75%       80.815000
max       98.760000
Name: OVERALL_LI, dtype: float64

📘 Interpretación del Resumen de Estadísticas
🔎 Información sobre la tasa de alfabetización
El resumen de estadísticas nos brinda información valiosa sobre la tasa general de alfabetización.

Por ejemplo, la media ayuda a identificar el centro del conjunto de datos.

Ahora sabemos que la tasa promedio de alfabetización es aproximadamente 73% en todos los distritos.

Esta información es útil por sí sola y también como base de comparación: conocer la media permite identificar qué distritos están significativamente por encima o por debajo de ella.

⚠️ Nota sobre valores faltantes
La función describe() excluye los valores faltantes (NaN) del análisis.

Por eso, el count (número de observaciones en la columna OVERALL_LI) es 634, menor que el número total de filas del dataset (680).

El manejo de valores faltantes es un tema complejo que está fuera del alcance de este curso.

📊 Uso de describe() en columnas categóricas
También puedes usar la función describe() en columnas con datos categóricos, como STATNAME.

Para una columna categórica, describe() devuelve:

count: número de observaciones no nulas.

unique: número de valores únicos.

top: valor más común (la moda).

freq: frecuencia del valor más común.

In [9]:
education_districtwise['STATNAME'].describe()

count         680
unique         36
top       STATE21
freq           75
Name: STATNAME, dtype: object



📘 Interpretación de categorías y funciones estadísticas en Python
🔎 Categorías en columnas categóricas
La categoría unique indica que hay 36 estados en el conjunto de datos.

La categoría top indica que STATE21 es el valor más común (la moda).

La categoría frequency muestra que STATE21 aparece en 75 filas, lo que significa que incluye 75 distritos diferentes.

👉 Esta información puede ser útil para determinar qué estados necesitan más recursos educativos según el número de distritos que contienen.

🧮 Funciones estadísticas en Python
La función describe() es práctica porque revela varias estadísticas clave de una sola vez.

Python también ofrece funciones separadas para calcular:

mean() → media.

median() → mediana.

std() → desviación estándar.

min() → valor mínimo.

max() → valor máximo.

📌 Ejemplo: anteriormente se usaron mean() y median() para detectar valores atípicos.

📏 Calcular el rango con min() y max()
El rango es la diferencia entre el valor más grande y el más pequeño de un conjunto de datos.

Fórmula:

Rango
=
max
−
min
En Python, puedes calcular el rango de la tasa de alfabetización de todos los distritos con:

In [10]:
range_overall_li = education_districtwise['OVERALL_LI'].max() - education_districtwise['OVERALL_LI'].min()
range_overall_li

np.float64(61.540000000000006)

Ese fragmento de código en Python con pandas lo que hace es calcular el rango de la columna OVERALL_LI (la tasa de alfabetización de cada distrito).

Paso a paso:
education_districtwise['OVERALL_LI'] → selecciona la columna de alfabetización.

.max() → obtiene el valor máximo de esa columna (el distrito con mayor tasa de alfabetización).

.min() → obtiene el valor mínimo de esa columna (el distrito con menor tasa de alfabetización).

max() - min() → resta ambos valores → eso es el rango, es decir, la diferencia entre el valor más alto y el más bajo.

range_overall_li → guarda ese resultado en una variable.

Al ejecutar range_overall_li, se muestra el valor calculado.

Ejemplo interpretado:
Si el distrito con mayor alfabetización tiene 95% y el de menor tiene 33.5%, entonces:

𝑅
𝑎
𝑛
𝑔
𝑜
=
95
−
33.5
=
61.5
Esto significa que la diferencia entre el mejor y el peor distrito en alfabetización es de 61.5 puntos porcentuales.

📘 Conclusión del laboratorio: Estadísticas descriptivas en Python
📊 Rango de tasas de alfabetización
El rango en las tasas de alfabetización de todos los distritos es de aproximadamente 61,5 puntos porcentuales.

Esta gran diferencia indica que algunos distritos tienen tasas de alfabetización mucho más altas que otros.

En análisis posteriores, podrás identificar qué distritos tienen las tasas más bajas.

Esta información ayuda al gobierno a comprender mejor la alfabetización a nivel nacional y a reforzar programas educativos exitosos.

✅ Conclusión del laboratorio
¡Felicitaciones! Has completado este laboratorio.

Ten en cuenta que puede que no aparezca una marca verde de verificación en la plataforma de Coursera.

Continúa tu progreso independientemente de la marca.

Haz clic en el ícono “guardar” en la parte superior del cuaderno para asegurarte de que tu trabajo quede registrado.

🚀 Próximos pasos
Ahora entiendes cómo calcular estadísticas descriptivas con Python.

A partir de aquí, puedes comenzar a usar estas estadísticas para explorar y resumir tus propios conjuntos de datos.

Esto te permitirá:

Detectar patrones.

Comparar valores.

Comunicar hallazgos de manera clara y efectiva.